sensors/dependencies/filter.go

```diff
@@ -205,13 +205,24 @@ filter:
 
 		case v1alpha1.JSONTypeString:
 			for _, value := range f.Value {
-				exp, err := regexp.Compile(value)
-				if err != nil {
-					return false, err
-				}
+				switch f.Comparator {
```

Comment:
A better logic might be:

case EqualTo: regex match
    
case NotEqualTo: !(regex match)

default: same as EqualTo

Reference to field filter logic in `resource` type event source (except `default`):
https://github.com/argoproj/argo-events/blob/master/eventsources/sources/resource/start.go#L288-L295

Reply:
Thanks @whynowy, I applied the changes.
I was not sure if would be better using regex or equallity, but at least it will be consistent with the rest.